File: fix_labels.js
const fs = require('fs');

function toSentenceCase(str) {
  // Simple sentence case: first letter uppercase, rest lowercase.
  // We should preserve acronyms like KRA, NSSF, SHA/NHIF, HQ.
  const acronyms = ["KRA", "PIN", "NSSF", "SHA/NHIF", "HQ"];
  return str.split(' ').map((word, i) => {
    if (acronyms.includes(word)) return word;
    if (i === 0) return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
    return word.toLowerCase();
  }).join(' ');
}

function processFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  
  // Regex to match <label ...>Text</label>
  content = content.replace(/<label\s+className="([^"]+)">([^<]+)<\/label>/g, (match, classes, text) => {
    // Make text white and remove uppercase
    let newClasses = classes
      .replace(/text-\[var\(--t[234]\)\]/, 'text-white')
      .replace(/text-\[1[01]px\]/, 'text-[12px]')
      .replace(/\buppercase\b/, '')
      .replace(/\btracking-wide\b/, '')
      .replace(/\s+/g, ' ')
      .trim();
    
    // Ensure text-white is there if not replaced
    if (!newClasses.includes('text-white')) {
      newClasses += ' text-white';
    }

    const newText = toSentenceCase(text);
    return `<label className="${newClasses}">${newText}</label>`;
  });

  fs.writeFileSync(filePath, content);
  console.log(`Processed ${filePath}`);
}

processFile('./src/components/OrganizationSetup/Foundation/OrganizationProfile.tsx');
processFile('./src/components/OrganizationSetup/SettingsPages.tsx');

File: fix_labels.py
import re
import os

files = [
    'src/components/OrganizationSetup/Foundation/OrganizationProfile.tsx',
    'src/components/OrganizationSetup/SettingsPages.tsx'
]

def to_sentence_case(text):
    words = text.split(' ')
    acronyms = ["KRA", "PIN", "NSSF", "SHA/NHIF", "HQ"]
    res = []
    for i, w in enumerate(words):
        if w in acronyms:
            res.append(w)
        elif i == 0:
            res.append(w.capitalize())
        else:
            res.append(w.lower())
    return ' '.join(res)

for file in files:
    if not os.path.exists(file): continue
    with open(file, 'r') as f:
        content = f.read()

    def replace_label(m):
        classes = m.group(1)
        text = m.group(2)
        classes = re.sub(r'\buppercase\b', '', classes)
        classes = re.sub(r'\btracking-wide\b', '', classes)
        classes = re.sub(r'text-\[var\(--t[234]\)\]', 'text-white', classes)
        if 'text-white' not in classes:
            classes += ' text-white'
        classes = ' '.join(classes.split())
        new_text = to_sentence_case(text)
        return f'<label className="{classes}">{new_text}</label>'

    new_content = re.sub(r'<label\s+className="([^"]+)">([^<]+)</label>', replace_label, content)
    
    with open(file, 'w') as f:
        f.write(new_content)
    print(f"Updated {file}")
